wordpress-mobile/WordPress-Android
# Jetpack App (Basics) - Add push notifications app key

Task #14468

This PR updates `NotificationUtils.registerDeviceForPushNotifications` to include `app_secret_key` in the `contentStruct`.
It sets the value as follows:
Jetpack App = `com.jetpack.android`
WordPress App = `org.wordpress.android`

It also replaces the deprecated call to get the token with the recommended one in facf4f8.

To test:

1. Fresh install the `wordPressJalapenoDebug` build flavor.
2. Add a breakpoint [here](https://github.com/wordpress-mobile/WordPress-Android/blob/3a2ff6d938707bf06de0c4e187d3aa589bdb8e24/WordPress/src/main/java/org/wordpress/android/ui/notifications/utils/NotificationsUtils.java#L105) and launch the app in the debug mode.
3. Notice that `app_secret_key` in the `contentStruct` is set to `org.wordpress.android`.

Note: 

1. Jetpack app needs to be added to the Firebase console before we start getting the registration token [here](https://github.com/wordpress-mobile/WordPress-Android/blob/develop/WordPress/src/main/java/org/wordpress/android/push/GCMRegistrationIntentService.java#L44) which is currently `null` for the Jetpack app and blocks the next step to send the app key to the server.

2. Push notifications should work using the `D60196` diff.

Merge Instructions:

- Wait for merge till both the apps are tested +ve for the push notifications. 
- Remove `Not Ready for Merge`. 
- Merge the PR.

## Regression Notes
1. Potential unintended areas of impact
N/A

2. What I did to test those areas of impact (or what existing automated tests I relied on)
N/A

3. What automated tests I added (or what prevented me from doing so)
N/A

PR submission checklist:
